pennylane/vqe/vqe.py

```diff
@@ -174,35 +191,39 @@ def simplify(self):
           (-1) [X0]
         + (1) [Y2]
         """
-        coeffs = []
+        data = []
         ops = []
 
-        for c, op in zip(self.coeffs, self.ops):
+        for i in range(len(self.ops)):  # pylint: disable=consider-using-enumerate
+            op = self.ops[i]
+            c = self.coeffs[i]
             op = op if isinstance(op, Tensor) else Tensor(op)
 
             ind = None
-            for i, other in enumerate(ops):
+            for j, other in enumerate(ops):
                 if op.compare(other):
-                    ind = i
+                    ind = j
                     break
 
             if ind is not None:
-                coeffs[ind] += c
-                if np.allclose([coeffs[ind]], [0]):
-                    del coeffs[ind]
+                data[ind] += c
+                if np.isclose(qml.math.toarray(data[ind]), np.array(0.0)):
+                    del data[ind]
                     del ops[ind]
             else:
                 ops.append(op.prune())
-                coeffs.append(c)
+                data.append(c)
```

Comment:
I added this last minute, because if `simplify` is used on an operator that is created already, we need to also update the internal parameters that are extracted for training by the tape - which are the ones stored under `self.data`. I hope we can tidy up this logic a bit soon.